src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityCallWarning.kt

```diff
@@ -0,0 +1,155 @@
+     * REGEX-TEST: §a✆ RING... RING... RING...
+     */
+    private val callRingPattern by ChocolateFactoryAPI.patternGroup.pattern(
+        "hoppity.call.ring",
+        "§a✆ (?:RING\\.{3} ?){1,3}(?:§r §r§2§l\\[PICK UP])?",
+    )
+
+    /**
+     * REGEX-TEST: eaf78cc9-260d-407f-b1df-efea83e5038a
+     * REGEX-TEST: 2bf7445f-2f86-406a-b629-deb5e6e03faa
+     */
+    private val cbUuidPattern by ChocolateFactoryAPI.patternGroup.pattern(
+        "hoppity.call.uuid",
+        "[a-f0-9]{8}-(?:[a-f0-9]{4}-){3}[a-f0-9]{12}",
+    )
+
+    /**
+     * REGEX-TEST: §e[NPC] §aHoppity§f: §b✆ §f§rWhat's up, §boBlazin§f?
+     */
+    private val pickupHoppityCallPattern by ChocolateFactoryAPI.patternGroup.pattern(
+        "hoppity.call.pickup",
+        "§e\\[NPC] §aHoppity§f: §b✆ §f§rWhat's up, .*§f\\?",
+    )
+
+    private val config get() = HoppityEggsManager.config.hoppityCallWarning
+    private var warningSound = SoundUtils.createSound("note.pling", 1f)
+    private var activeWarning = false
+    private var nextWarningTime = 0L
+    private var finalWarningTime = 0L
+    private const val CALL_LENGTH_MS = 7000
+    private var acceptUUID: String? = null
+
+    @SubscribeEvent
+    fun onKeyPress(event: LorenzKeyPressEvent) {
+        if (acceptUUID == null || config.acceptHotkey == Keyboard.KEY_NONE || config.acceptHotkey != event.keyCode) return
+        HypixelCommands.cb(acceptUUID!!)
+        acceptUUID = null
+    }
+
+    @SubscribeEvent
+    fun onConfigLoad(event: ConfigLoadEvent) {
+        val soundProperty = config.hoppityCallSound
+        ConditionalUtils.onToggle(soundProperty) {
+            warningSound = SoundUtils.createSound(soundProperty.get(), 1f)
+        }
+        nextWarningTime = 0L
+        finalWarningTime = 0L
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    fun onChat(event: LorenzChatEvent) {
+        if (callRingPattern.matches(event.message) && acceptUUID == null) extractPickupUuid(event)
+        if (!isEnabled()) return
+        if (initHoppityCallPattern.matches(event.message)) startWarningUser()
+        if (pickupHoppityCallPattern.matches(event.message)) stopWarningUser()
+    }
+
+    @SubscribeEvent
+    fun onTick(event: SecondPassedEvent) {
+        if (!isEnabled()) return
+        if (!activeWarning) return
+        if (nextWarningTime == 0L || finalWarningTime == 0L) return
+        val currentTime = System.currentTimeMillis()
+        if (currentTime >= nextWarningTime) {
+            SoundUtils.repeatSound(100, 10, warningSound)
+            nextWarningTime = currentTime + 100
+        }
+        if (currentTime >= finalWarningTime) stopWarningUser()
+    }
+
+    @SubscribeEvent
+    fun onRender(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        if (!activeWarning) return
+        val minecraft = Minecraft.getMinecraft()
+        val alpha = ((2 + sin(System.currentTimeMillis().toDouble() / 1000)) * 255 / 4).toInt().coerceIn(0..255)
```

Comment:
please extract this line into multiple lines, give them useful variable names and eventually even comments to increase maintainability.